File: drowsiness-detection-app/src/features/auth/services/authApi.ts
import apiClient from '../../../lib/api/client';
import type { LoginCredentials, AuthResponse, User } from '../../../types';

export const authApi = {
  // Login
  login: async (credentials: LoginCredentials): Promise<AuthResponse> => {
    const response = await apiClient.post<AuthResponse>('/auth/login', credentials);
    return response.data;
  },

  // Obtener usuario actual
  getCurrentUser: async (): Promise<User> => {
    const response = await apiClient.get<User>('/auth/me');
    return response.data;
  },

  // Logout
  logout: async (): Promise<void> => {
    await apiClient.post('/auth/logout');
  },

  // Refresh token
  refreshToken: async (refreshToken: string): Promise<{ access_token: string; token_type: string }> => {
    const response = await apiClient.post('/auth/refresh', {
      refresh_token: refreshToken,
    });
    return response.data;
  },
};

export default authApi;

File: DriverDrowsinessDetectorApp/app/src/main/java/com/example/driverdrowsinessdetectorapp/data/remote/api/AuthApi.kt
package com.example.driverdrowsinessdetectorapp.data.remote.api

import com.example.driverdrowsinessdetectorapp.data.remote.dto.request.LoginRequest
import com.example.driverdrowsinessdetectorapp.data.remote.dto.response.LoginResponse
import com.example.driverdrowsinessdetectorapp.data.remote.dto.response.UserResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface AuthApi {
    @POST("/api/v1/auth/login")
    suspend fun login(@Body request: LoginRequest): LoginResponse
    
    @POST("/api/v1/auth/logout")
    suspend fun logout()
    
    @GET("/api/v1/users/me")
    suspend fun getCurrentUser(): UserResponse
}